ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/NullScanTaskDispatcher.java

```diff
@@ -159,21 +165,46 @@ private void processTableScans(MapWork work, Set<TableScanOperator> tableScans)
           } else {
             managedEmptyPathMap.put(path, false);
           }
+        } else {
+          numberOfUnmanagedPaths++;
         }
       }
     }
 
+    int fetcherPoolParallelism = Math.min(MAXIMUM_ASYNC_STATUS_FETCHER_COUNT, (int)Math.sqrt(numberOfUnmanagedPaths));
+    ForkJoinPool pool = null;
```

Comment:
Can use regular ExecutorService since it is not doing something very similar to ForkJoin.